src/services/edusharing/logic/connector.js

```diff
@@ -0,0 +1,248 @@
+		}
+		const options = {
+			method: 'POST',
+			// This will be changed later with a qs where sorting, filtering etc is present.
+			// eslint-disable-next-line max-len
+			url: `${process.env.ES_DOMAIN}/edu-sharing/rest/search/v1/queriesV2/mv-repo.schul-cloud.org/mds/ngsearch/?contentType=${contentType}&skipCount=${skipCount}&maxItems=${maxItems}&sortProperties=${sortProperties}&sortProperties=cm%3Amodified&sortAscending=${sortAscending}&sortAscending=false&propertyFilter=${propertyFilter}&`,
+			headers: {
+				...EduSharingConnector.headers,
+				cookie: this.authorization,
+			},
+			body: JSON.stringify({
+				criterias: [
+					{ property: 'ngsearchword', values: [`${searchWord}`] }
+				],
+				facettes: ['cclom:general_keyword']
+			}),
+			timeout: REQUEST_TIMEOUT,
+		};
+
+		// code below includes params in seperate key/value, but does not work.
+
+		/* const options2 = {
+			method: 'POST',
+			// This will be changed later with a qs where sorting, filtering etc is present.
+			// eslint-disable-next-line max-len
+			//url: `${process.env.ES_DOMAIN}/edu-sharing/rest/search/v1/queriesV2/mv-repo.schul-cloud.org/mds/ngsearch/?contentType=${contentType}&skipCount=${skipCount}&maxItems=${maxItems}&sortProperties=score&sortProperties=cm%3Amodified&sortAscending=false&sortAscending=false&propertyFilter=-all-&`,
+			uri: `${process.env.ES_DOMAIN}/edu-sharing/rest/search/v1/queriesV2/mv-repo.schul-cloud.org/mds/ngsearch/`,
+			qs: {
+				contentType,
+				skipCount,
+				sortProperties,
+				sortAscending,
+				propertyFilter,
+			},
+
+			headers: {
+				...EduSharingConnector.headers,
+				cookie: this.authorization,
+			},
+			body: JSON.stringify({
+				criterias: [
+					{ property: 'ngsearchword', values: [`${searchWord}`] }
+				],
+				facettes: ['cclom:general_keyword']
+			}),
+			timeout: REQUEST_TIMEOUT
+		}; */
+
+		const eduResponse = await request(options);
+		let parsed = JSON.parse(eduResponse);
+
+		// provided by client eg data.query.filterOptions
+		const filterOptions = {
+			mimetype: ['text/html', 'image/jpeg'],
+			provider: ['BauhausMaterial.de', 'München educationcenter', 'Khan Academy']
+		}
+
+		// checks if user has set filter options
+		if (!!Object.values(filterOptions).length) {
+			parsed = filterResult(parsed, filterOptions)
+		}
+
+		function filterResult(obj, options) {
+			let result;
+			// checks if user has set type filter
+			if (options.mimetype.length) {
+				result = filterMime(obj, options.mimetype)
+			}
+			// checks if user has set provider filter
+			if (options.provider.length) {
+				result = filterProvider(obj, options.provider)
+			}
+			return result
+		}
+
+		// filter away everything buy selected mimetype
+		function filterMime(obj, mimetypes) {
+			let result = obj.nodes
+			mimetypes.forEach(type => { result = result.filter(n => n.mimeType === type) })
+			return result
+		}
+
+		// filter away everything buy selected providers
+		function filterProvider(obj, providers) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['providers' is defined but never used. Allowed unused args must match /app|req|res|next|options|params/u.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=4791331)